### ISSUE TITLE
Gemma3 weight is negative

### ISSUE THREAD

**Issue by sempervictus** [EricLBuehler/candle-vllm#310]
```shell
$ candle-vllm --p 12099 --m google/gemma-3-4b-it --isq q4k --log --mem 8192 --verbose
```
same for `q4_1`
```rust
2025-09-19 23:53:38 WARN candle_vllm subprocess rank 0 started!
2025-09-19 23:53:38 WARN candle_vllm::openai::communicator command channel is built!
2025-09-19 23:53:38 WARN candle_vllm::openai::communicator command channel is built!
2025-09-19 23:53:38 WARN candle_vllm::openai::pipelines::pipeline create nccl comm channel rank 0, shards 1, id Id { id: ncclUniqueId { internal: [123, 27, 102, -63, 35, 83, 124, 94, 2, 0, -49, -91, -64, -88, 121, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] } }
2025-09-19 23:53:39 WARN candle_vllm::openai::pipelines::pipeline nccl comm created for rank 0
2025-09-19 23:54:00 WARN candle_vllm::backend::progress all ranks finished model loading!
2025-09-19 23:54:00 WARN candle_vllm::openai::pipelines::pipeline Done loading.
2025-09-19 23:54:01 WARN candle_vllm::openai::pipelines::pipeline stop_token_ids [1, 106]
2025-09-19 23:54:01 WARN candle_vllm Maximum Model Length (affected by `--mem` (kvcache-mem-gpu) and the number of ranks):
-> Batch 1: 61664
-> Batch 8: 7708
2025-09-19 23:54:01 WARN candle_vllm Server started at http://0.0.0.0:12099.
2025-09-19 23:54:04 WARN candle_vllm::openai::pipelines::llm_engine New Request with length 1257 tokens (cmpl-6767aaa7-92f0-400b-a53d-268da7305254).

thread 'tokio-runtime-worker' (519525) panicked at src/openai/logits_processor.rs:198:51:
called `Result::unwrap()` on an `Err` value: A weight is negative, too large or not a valid number
   0: candle_core::error::Error::bt
   1: candle_vllm::openai::logits_processor::LogitsProcessor::sample_multinomial
   2: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &F>::call_mut
   3: rayon::iter::plumbing::Folder::consume_iter
   4: rayon::iter::plumbing::bridge_producer_consumer::helper
   5: rayon::iter::collect::collect_with_consumer
   6: rayon::iter::extend::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend
   7: candle_vllm::openai::logits_processor::LogitsProcessor::sample
   8: candle_vllm::openai::pipelines::pipeline::DefaultPipeline::sample
   9: candle_vllm::openai::pipelines::llm_engine::LLMEngine::generate_once
  10: rayon::iter::plumbing::Folder::consume_iter
  11: rayon::iter::plumbing::bridge_producer_consumer::helper
  12: rayon::iter::collect::collect_with_consumer
  13: tokio::runtime::handle::Handle::block_on
  14: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
  15: tokio::runtime::task::raw::poll
  16: tokio::runtime::blocking::pool::Inner::run
  17: std::sys::backtrace::__rust_begin_short_backtrace
  18: core::ops::function::FnOnce::call_once{{vtable.shim}}
  19: std::sys::thread::unix::Thread::new::thread_start
  20: <unknown>
  21: <unknown>

```

**Comment by guoqingbao** [comment]
> $ candle-vllm --p 12099 --m google/gemma-3-4b-it --isq q4k --log --mem 8192 --verbose
> same for `q4_1`
> 
> 2025-09-19 23:53:38 WARN candle_vllm subprocess rank 0 started!
> 2025-09-19 23:53:38 WARN candle_vllm::openai::communicator command channel is built!
> 2025-09-19 23:53:38 WARN candle_vllm::openai::communicator command channel is built!
> 2025-09-19 23:53:38 WARN candle_vllm::openai::pipelines::pipeline create nccl comm channel rank 0, shards 1, id Id { id: ncclUniqueId { internal: [123, 27, 102, -63, 35, 83, 124, 94, 2, 0, -49, -91, -64, -88, 121, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] } }
> 2025-09-19 23:53:39 WARN candle_vllm::openai::pipelines::pipeline nccl comm created for rank 0
> 2025-09-19 23:54:00 WARN candle_vllm::backend::progress all ranks finished model loading!
> 2025-09-19 23:54:00 WARN candle_vllm::openai::pipelines::pipeline Done loading.
> 2025-09-19 23:54:01 WARN candle_vllm::openai::pipelines::pipeline stop_token_ids [1, 106]
> 2025-09-19 23:54:01 WARN candle_vllm Maximum Model Length (affected by `--mem` (kvcache-mem-gpu) and the number of ranks):
> -> Batch 1: 61664
> -> Batch 8: 7708
> 2025-09-19 23:54:01 WARN candle_vllm Server started at http://0.0.0.0:12099.
> 2025-09-19 23:54:04 WARN candle_vllm::openai::pipelines::llm_engine New Request with length 1257 tokens (cmpl-6767aaa7-92f0-400b-a53d-268da7305254).
> 
> thread 'tokio-runtime-worker' (519525) panicked at src/openai/logits_processor.rs:198:51:
> called `Result::unwrap()` on an `Err` value: A weight is negative, too large or not a valid number
>    0: candle_core::error::Error::bt
>    1: candle_vllm::openai::logits_processor::LogitsProcessor::sample_multinomial
>    2: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &F>::call_mut
>    3: rayon::iter::plumbing::Folder::consume_iter
>    4: rayon::iter::plumbing::bridge_producer_consumer::helper
>    5: rayon::iter::collect::collect_with_consumer
>    6: rayon::iter::extend::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend
>    7: candle_vllm::openai::logits_processor::LogitsProcessor::sample
>    8: candle_vllm::openai::pipelines::pipeline::DefaultPipeline::sample
>    9: candle_vllm::openai::pipelines::llm_engine::LLMEngine::generate_once
>   10: rayon::iter::plumbing::Folder::consume_iter
>   11: rayon::iter::plumbing::bridge_producer_consumer::helper
>   12: rayon::iter::collect::collect_with_consumer
>   13: tokio::runtime::handle::Handle::block_on
>   14: <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
>   15: tokio::runtime::task::raw::poll
>   16: tokio::runtime::blocking::pool::Inner::run
>   17: std::sys::backtrace::__rust_begin_short_backtrace
>   18: core::ops::function::FnOnce::call_once{{vtable.shim}}
>   19: std::sys::thread::unix::Thread::new::thread_start
>   20: <unknown>
>   21: <unknown>

I think that's the accuracy problem on V100, have you tetsted q8_0?

**Comment by guoqingbao** [comment]
Solved at #406
